addons/point_of_sale/models/account_journal.py

```diff
@@ -27,3 +28,55 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
     def onchange_type(self):
         if self.type not in ['bank', 'cash']:
             self.journal_user = False
+
+    @api.multi
+    def write(self, vals):
+        """ Prevent to archive journals that are still used in an opened session """
+        if not vals.get('active', True):
+            config_ids = self.env['pos.config'].search([
+                '|',
+                    ('journal_id', 'in', self.ids),
+                    ('journal_ids', 'in', self.ids)
+            ]).ids
+            if config_ids:
+                running_sessions = self.env['pos.session'].search([
+                    ('state', '!=', 'closed'),
+                    ('config_id', 'in', config_ids)
+                ])
+                if running_sessions:
+                    # find the problematic journal back from the one in self
+                    journals = running_sessions.mapped('config_id.journal_id')
+                    journals |= running_sessions.mapped('config_id.journal_ids')
+                    journals = journals & self
+                    raise UserError(
+                        _("You cannot archive the journal %s as it is still used in the PoS sessions %s.")
```

Comment:
is the pos session name really the clearest way to guide the user? Personally I think it's clearer to use the pos.config name instead, since that the name you see in the dashboard...